app/domain/authentication/authn_jwt/jwt_fetch_jwks_uri_signing_key.rb

```diff
@@ -0,0 +1,51 @@
+require 'uri'
+require 'net/http'
+
+module Authentication
+  module AuthnJwt
+
+    JwtFetchJwksUriSigningKey = CommandClass.new(
+      dependencies: {
+        logger: Rails.logger,
+        discover_identity_provider: OAuth::DiscoverIdentityProvider.new,
+        fetch_secrets: Conjur::FetchRequiredSecrets.new
+      },
+      inputs: %i[authenticator_input]
+    ) do
+      def call
+        jwks_uri
+        fetch_jwks_keys
+      end
+
+      private
+
+      def jwks_uri
+        @logger.debug(LogMessages::Authentication::AuthnJwt::JwksUriResourceNameConfiguration.new(resource_id))
+        @jwks_uri_secret ||= @fetch_secrets.(resource_ids: [resource_id])
+        @jwks_uri ||=@jwks_uri_secret[resource_id]
+      end
+
+      def resource_id
+        "#{@authenticator_input.account}:variable:conjur/#{@authenticator_input.authenticator_name}/#{@authenticator_input.service_id}/#{JWKS_URI_RESOURCE_NAME}"
+      end
```

Comment:
2 points:
1. please change the method name to `jwks_uri_resource_id` to be more explicit.
2. i asked this in another PR but never really got an answer - why aren't we using `FetchAuthenticatorSecrets` instead of `FetchRequiredSecrets`? i think it has some logic that will save you some code.